Source/ASRunLoopQueue.mm

```diff
@@ -342,22 +352,15 @@ @implementation ASCATransactionQueue
 // but after most other scheduled work on the runloop has processed.
 static int const kASASCATransactionQueueOrder = 1000000;
 
-+ (ASCATransactionQueue *)sharedQueue NS_RETURNS_RETAINED
-{
-  static dispatch_once_t onceToken;
-  static ASCATransactionQueue *sharedQueue;
-  dispatch_once(&onceToken, ^{
-    sharedQueue = [[ASCATransactionQueue alloc] init];
-  });
-  return sharedQueue;
-}
-
 - (instancetype)init
 {
   if (self = [super init]) {
-    _runLoop = CFRunLoopGetMain();
-    NSPointerFunctionsOptions options = NSPointerFunctionsStrongMemory;
-    _internalQueue = [[NSPointerArray alloc] initWithOptions:options];
+    _internalQueueHashSet = CFSetCreateMutable(NULL, 0, NULL);
+    
+    // This is going to be a very busy queue – every node in the preload range will enter this queue.
+    // Save some time on first render by reserving space up front.
+    _internalQueue.reserve(64);
```

Comment:
What does this mean? (64 bits for each pointer?)

Reply:
No it just tells the vector to allocate space in advance for 64 items.

By default, when you add items to a std::vector, it will first reserve 1 spot. Then if you add another, it will reserve 2, move the 1, and delete the old buffer. Then if you add another, it'll reserve 4, move the 2, and delete the old buffer and so-on. To save all these allocations, we skip over the 1 2 4 8 16 and 32 steps.